### PR TITLE
For static hosting, first arg is root path

### PR DESCRIPTION
**Pull request by ewalk153**
Code assumed first argument to static workers is the root path, but the example incorrectly showed it to be a directory mapping.

This fix supports both: it adds “/“ if there is no mapping or else it accepts a mapping now. 

This also changes the example from the Procfile only pass the root directory. 

## Diagnosis

Error message from deploying the current static example:
```
remote: Error too many values to unpack (expected 2) in static path spec: should be /url1:path1[,/url2:path2], ignoring.
```

Static Procfile line parsing prepends the root path automatically.

https://github.com/piku/piku/blob/a433a68a1dd2d962033f44a818920a4c56899399/piku.py#L886

So the static path variable looks like this:
`['/::public', '/somepath:somedir/']`

Trailing "/" required after public or the path returns 403 errors when appending a second path.

Testing instructions:
You'll need to alter the githome var in piku to use this branches' `piku init` 
https://github.com/piku/piku/blob/a433a68a1dd2d962033f44a818920a4c56899399/piku#L12
becomes:
```
githome="https://raw.githubusercontent.com/ewalk153/piku/correct-static-assets-example/"
```
to test against this branch.

### Alternate approach

We could test to see if there is a colon in the first argument and only prepend the `/:` if it's not present. This should be backwards compatible as the current behavior is broken (breaks the deploy)

### CODE REVIEW

**Comment by ewalk153** [comment]
Looks like the python 3.6 for x64 is no longer available:
```
Error: Version 3.6 with arch x64 not found
```

**Comment by chr15m** [comment]
How confusing. Thanks for finding this bug. We really need a bunch of test cases set up to see how it parses different static path settings. Let me have a think about this one.

> We could test to see if there is a colon in the first argument and only prepend the /: if it's not present.

I do like this solution though. I think the current example init should probably work as is rather than changing it.

File: piku.py [hunk 1]
```diff
@@ -884,14 +884,14 @@ def spawn_app(app, deltas={}):
             # prepend static worker path if present
             if 'static' in workers:
                 stripped = workers['static'].strip("/").rstrip("/")
-                static_paths = "/:" + (stripped if stripped else ".") + "/" + ("," if static_paths else "") + static_paths
+                static_paths = ("/" if stripped[0:1] == ":" else "/:") + (stripped if stripped else ".") + "/" + ("," if static_paths else "") + static_paths
```
Comment (ewalk153) [inline piku.py:887]: This only adds a leading "/:" if it's absent.

File: piku.py [hunk 2]
```diff
@@ -884,14 +884,14 @@ def spawn_app(app, deltas={}):
...
             if len(static_paths):
                 try:
                     items = static_paths.split(',')
                     for item in items:
                         static_url, static_path = item.split(':')
                         if static_path[0] != '/':
-                            static_path = join(app_path, static_path)
+                            static_path = join(app_path, static_path).rstrip("/") + "/"
```
Comment (ewalk153) [inline piku.py:894]: What do you folks think about normalizing the trailing slash?

Otherwise the current Procfile example:
```
static: /:public,/somepath:somedir
```
expands to:
```
  location / {
      sendfile on;
      sendfile_max_chunk 1m;
      tcp_nopush on;
      directio 8m;
      aio threads;
      alias /home/piku/.piku/apps/public-example/public;
      try_files $uri $uri.html $uri/ =404;
  }
```
`/home/piku/.piku/apps/public-example/public` misses the `/` after public and nginx this return 403 for requests to the root dir.

**Comment by ewalk153** [comment]
I extracted the logic into a small test to verify the behavior:
https://gist.github.com/ewalk153/ac2fe326a4b379ae12a5cba81e4f3856

tl;dr: this correctly handles trailing `/` chracters.

**Comment by rcarmo** [comment]
This looks reasonably OK to me (although it was a bit of a challenge to follow)...

**Comment by chr15m** [comment]
Same here! Thanks for looking into this @ewalk153 - hopefully it will make the static path serving more robust to different configurations.
